### ISSUE TITLE
Add Support for PythonMonkey & stPyV8

### ISSUE THREAD

**Issue by emmyarty** [emmyarty/dxsvelte#14]
I encountered this today

https://github.com/Distributive-Network/PythonMonkey

Can be of value to this project

_Originally posted by @baseplate-admin in https://github.com/emmyarty/dxsvelte/discussions/4#discussioncomment-6766170_

**Comment by almahdi404** [comment]
**PythonMonkey** has an [issue](https://github.com/Distributive-Network/PythonMonkey/issues/163) with `threading`

Currently doesn't work with **Django**

**Comment by Hansimov** [comment]
Use python `subprocess` as a workaround. Just take care when parsing all kinds of quotes.

An example:

```py
import subprocess

def eval_js_codes(js_codes):
    js_codes_to_eval = js_codes.replace('"', '\\"')
    js_codes_to_eval_in_python = f"from pythonmonkey import eval; eval('{js_codes_to_eval}')"
    pm_cmd = f'python -c "{js_codes_to_eval_in_python}"'
    output = subprocess.getoutput(pm_cmd)
    return output

js_codes = 'console.log("Hello World!")'
eval_js_codes(js_codes)

# python -c "from pythonmonkey import eval; eval('console.log(\"Hello World!")')"
```

Output:

```bash
Hello World!
```

**Comment by baseplate-admin** [comment]
@almahdi404, আমার মনে হয় না যে `PythonMonkey` কাজ করবে। ডেভেলপমেন্ট অনেক ধীর গতিতে হচ্ছে। এর থেকে ভাল হয় [`mozjs`](https://github.com/servo/mozjs) ব্যবহার করা। এটা `thread-safe`. কিন্তু তার `wheel` নাই। এজন্য আমাদের বর্তমান `py-mini-racer` সবচেয়ে ভাল dependency হিসাবে। 

----

@emmyarty, let's close this issue. The `spidermonkey` is not developed at a pace that's suitable for our scope ( like if we want to support latest ECMAScript ). So it's better to look for alternatives.

**Comment by almahdi404** [comment]
I am building a framework for django, which is able to render alpinejs syntax on the server. I planed to use `PythonMonkey` but eventually used `py-mini-racer` for it. Unfortunately `py-mini-racer` has not been updated for 2 years.
A better alternative to `py-mini-racer` would be [STPyV8](https://github.com/cloudflare/stpyv8). But its not available in **pypi**.

**Comment by baseplate-admin** [comment]
> I am building a framework for django, which is able to render alpinejs syntax on the server. I planed to use PythonMonkey but eventually used py-mini-racer for it. Unfortunately py-mini-racer has not been updated for 2 years.

About this, did you try the alternative proposed in the last comment? https://github.com/Distributive-Network/PythonMonkey/issues/163#issuecomment-1855657379

> A better alternative to py-mini-racer would be [STPyV8](https://github.com/cloudflare/stpyv8). But its not available in pypi.

Seems better maintained too, i just [commented](https://github.com/cloudflare/stpyv8/issues/81#issuecomment-1855783882) on their latest discussion.

**Comment by almahdi404** [comment]
> About this, did you try the alternative proposed in the last comment? [Distributive-Network/PythonMonkey#163 (comment)](https://github.com/Distributive-Network/PythonMonkey/issues/163#issuecomment-1855657379)

No. I replied to that here(https://github.com/Distributive-Network/PythonMonkey/issues/163#issuecomment-1855852206).

**Comment by baseplate-admin** [comment]
so yea overall JS support on python is pretty poor :')
